CRM/Contribute/Form/UpdateSubscription.php

```diff
@@ -224,15 +225,15 @@ public function postProcess() {
     $params['subscriptionId'] = $this->_subscriptionDetails->subscription_id;
     $updateSubscription = TRUE;
     if ($this->_paymentProcessorObj->supports('changeSubscriptionAmount')) {
-      $updateSubscription = $this->_paymentProcessorObj->changeSubscriptionAmount($message, $params);
-    }
-    if (is_a($updateSubscription, 'CRM_Core_Error')) {
-      CRM_Core_Error::displaySessionError($updateSubscription);
-      $status = ts('Could not update the Recurring contribution details');
-      $msgTitle = ts('Update Error');
-      $msgType = 'error';
+      try {
+        $updateSubscription = $this->_paymentProcessorObj->changeSubscriptionAmount($message, $params);
+      }
+      catch (\Civi\Payment\Exception\PaymentProcessorException $e) {
```

Comment:
@artfulrobot I meant, and I think that we should also catch the CRM_Core_Error here